### ISSUE TITLE
Question about PI model implementation

### ISSUE THREAD

**Issue by SkieyFly** [huggingface/lerobot#2863]
### Ticket Type

❓ Technical Question

### Environment & System Info

```Shell

```

### Description

While reading the PI model implementation, I noticed several differences between current implementation and OpenPI's implementation.

- The loss is truncated to `original_action_dim`. 
lerobot: https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi05/modeling_pi05.py#L1252-L1257.

- The state is already padded **before** being discretized.
lerobot: https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi05/processor_pi05.py#L71-L77. 
openpi: https://github.com/Physical-Intelligence/openpi/blob/main/src/openpi/training/config.py#L114-L138.

I wanted to check whether I might have overlooked something. Could these differences be related to the training loss instability I’m observing?

### Relevant logs or stack trace

```Shell

```

### Checklist

- [ ] I have searched existing tickets to ensure this isn't a duplicate.
- [ ] I am using the latest version of the `main` branch.
- [ ] I have verified this is not an environment-specific problem.

**Comment by pkooij** [comment]
Hi @SkieyFly, yes these are valid differences which I think are worth it to modify. I can create a PR to instead of truncating modify the mask and remove the state padding? Or would you be open to create a PR?

**Comment by SkieyFly** [comment]
Hi @pkooij,
I've created a PR (#2891) but I'm afraid I don't have too much time to do full sanity check. Can you help double check them? I've also provided some links that you can refer to. Thanks a lot!
